flowmachine/flowmachine/features/subscriber/home_location_monthly.py

```diff
@@ -0,0 +1,198 @@
+from typing import List, Union
+from flowmachine.core.spatial_unit import AnySpatialUnit
+from datetime import datetime, timedelta
+from flowmachine.core.query import Query
+from dateutil.relativedelta import relativedelta
+from dateutil.rrule import rrule, DAILY
+from copy import copy
+
+from flowmachine.features.subscriber.daily_location import daily_location
+from flowmachine.features.subscriber.modal_location import ModalLocation
+from flowmachine.features.subscriber.unique_active_subscribers import (
+    UniqueActiveSubscribers,
+    ActiveSubscribers,
+)
+from flowmachine.features.subscriber.last_location import LastLocation
+from flowmachine.features.subscriber.location_visits import LocationVisits
+
+
+class HomeLocationMonthly(Query):
+
+    """For each subscriber, if their modal last location was their last location on more than X days this month
+    (choose X=15 here), then choose this as their home location. If not, but it was their last location on more than Y
+     days this month (Y<X; choose Y=10 here) AND it is the same as their “reference location” (in our case reference
+     location is modal location and last location on more than X days last month), then choose this as their home
+     location. Otherwise set their home location to “unlocatable” (a special value, not NULL)."""
+
+    def __init__(
+        self,
+        window_start: Union[str, datetime],
+        window_stop: Union[str, datetime],
+        agg_unit: AnySpatialUnit,
+        home_this_month: int,
+        home_last_month: int,
+        ref_location: Union["HomeLocationMonthly", None] = None,
+        events_tables=None,
+        modal_lookback=40,
+        active_days=4,
+        interval=7,
+    ):
+
+        self.window_start = window_start
+        self.window_stop = window_stop
+        self.ref_location = ref_location
+        self.agg_unit = agg_unit
```

Comment:
I'd prefer to call this parameter 'spatial_unit', I think, for consistency with the rest of flowmachine. It's not critical here, because we're not using GeoDataMixin (which explicitly looks for a `spatial_unit` attribute), but it's good to be consistent. (I know we use 'aggregation_unit' for the equivalent parameters exposed through the API, and we may want to harmonise these at some point, but I think that harmonisation would be easier if flowmachine and flowapi are separately self-consistent).